packages/cloudflare/src/cli/build/build-worker.ts

```diff
@@ -134,6 +135,7 @@ const CustomRequest = class extends globalThis.Request {
 };
 globalThis.Request = CustomRequest;
 Request = globalThis.Request;
+globalThis.__dangerous_ON_edge_converter_returns_request = true;
```

Comment:
what is this for? should we document it with a comment or something?

Reply:
The ona [middleware handler](https://github.com/opennextjs/opennextjs-aws/blob/f7758ac91d29e3cedb86301ecc6ab4e7af581024/packages/open-next/src/adapters/middleware.ts) takes an `InternalEvent` and returns an `InternalResult`.

Cloudflare handlers takes a `Request` and return a `Response`.

To adapt those types, Open Next uses a converter. By default [the edge converter](https://github.com/opennextjs/opennextjs-aws/blob/f7758ac91d29e3cedb86301ecc6ab4e7af581024/packages/open-next/src/overrides/converters/edge.ts) will convert the `Request` to an `InternalEvent` (`convertFrom` is called before the middleware handler) and once the middleware handler returns, `convertTo` is called to convert the `InternalResult` to a `Response`.

In the current cloudflare implementation we do not always want to return a `Response`. We only want want to `Response` when the middleware generates one (redirects or plain response). In all others cases we will send the `Request` to the next server.

`globalThis.__dangerous_ON_edge_converter_returns_request` instructs the edge converter to returns the request before trying to fetch the response from an external next server - our next server is local.